### PR TITLE
refactor: Move the logic for detecting supported browsers for a wallet to core package

### PR DESCRIPTION
**Pull request by kujtimprenkuSQA**
# Description

This PR is about moving the logic for detecting the browser which an injected(extension) wallet supports to the `core` package so we can reuse it in other packages.

The initial work was done as part of the previous PR: https://github.com/near/wallet-selector/pull/735

This introduces a new helper function `isCurrentBrowserSupported` that accepts the list of browsers that a wallet supports, the goal is to show a wallet only on the browsers that is supported.

The injected wallets will need to update the code to something like this in the setup function:

```ts
const mobile = isMobile();
const isSupported = isCurrentBrowserSupported(["chrome", "edge-chromium", "opera"]); 

if (mobile || !isSupported ) {
   return null;
}
```

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

# Type of change.

- [ ] FIX - a PR of this type patches a bug.
- [ ] FEATURE - a PR of this type introduces a new feature.
- [ ] BUILD - a PR of this type introduces build changes.
- [ ] CI - a PR of this type introduces CI changes.
- [ ] DOCS - a PR of this type introduces DOCS improvement.
- [ ] STYLE - a PR of this type introduces style changes.
- [x] REFACTOR - a PR of this type introduces refactoring.
- [ ] PERFORMANCE - a PR of this type introduces performance changes.
- [ ] TEST - a PR of this type adds more tests.
- [ ] CHORE - a PR introduces other changes than the specified above.

### CODE REVIEW

**Review by DamirSQA — DISMISSED** [review]

File: packages/welldone-wallet/src/lib/welldone.ts [hunk 1]
```diff
@@ -361,9 +360,10 @@ export function setupWelldoneWallet({
   deprecated = false,
 }: WelldoneWalletParams = {}): WalletModuleFactory<InjectedWallet> {
   return async () => {
-    const isSupported =
-      !isMobile() && isSupportBrowser(["chrome", "edge-chromium", "opera"]);
-    if (!isSupported) {
+    const mobile = isMobile();
+    const supported = isBrowserSupported(["chrome", "edge-chromium", "opera"]);
```
Comment (hcho112) [inline packages/welldone-wallet/src/lib/welldone.ts:364]: can we keep the name `isSupported` as it is boolean

**Review by hcho112 — CHANGES_REQUESTED** [review]
minor change request. In terms of name of the helper function, I'm open to suggestion as well.

File: packages/core/src/lib/helpers/isBrowserSupported.ts [hunk 2]
```diff
@@ -99,7 +99,7 @@ const matchUserAgent = (ua: string): UserAgentMatch => {
   );
 };
 
-export const isSupportBrowser = (browers: Array<Browser>): boolean => {
+export const isBrowserSupported = (browsers: Array<Browser>): boolean => {
```
[the remark this answers is not in the document]
Comment (kujtimprenkuSQA) [reply packages/core/src/lib/helpers/isBrowserSupported.ts:102]: Yeah, isCurrentBrowserSupported is much better.

File: packages/welldone-wallet/src/lib/welldone.ts (hunk 1 above)
Comment (kujtimprenkuSQA) [reply packages/welldone-wallet/src/lib/welldone.ts:364]: Alright will change it 👍

**Review by DamirSQA — APPROVED** [review]

**Review by hcho112 — APPROVED** [review]
LGTM

**Pull request merged**